ksql-engine/src/main/java/io/confluent/ksql/function/udaf/KudafAggregator.java

```diff
@@ -62,11 +62,11 @@ public GenericRow apply(String s, GenericRow rowValue, GenericRow aggRowValue) {
       GenericRow mergedRow = new GenericRow(columns);
 
       aggValToValColumnMap.forEach((columnIndex, value) -> {
-        if (aggRowOne.getColumns().get(value).toString().length() > 0) {
-          mergedRow.getColumns().set(columnIndex, aggRowOne.getColumns()
+        if (aggRowOne.getColumns().get(value) == null) {
+          mergedRow.getColumns().set(columnIndex, aggRowTwo.getColumns()
```

Comment:
The logic seems to have been inverted in this change set. Was that intentional? Was the previous behavior a bug?

Reply:
yes it was intentional - previously the default value for the columns was `""` hence the string length check. I've changed the default value to be `null`